src/Arcus.WebApi.Security/Authentication/SharedAccessKey/SharedAccessKeyAuthenticationFilter.cs

```diff
@@ -106,29 +129,14 @@ public async Task OnAuthorizationAsync(AuthorizationFilterContext context)
 
         private async Task<string[]> GetAuthorizationSecretAsync(AuthorizationFilterContext context)
         {
-            var userDefinedSecretProvider = context.HttpContext.RequestServices.GetService<ISecretProvider>();
-            if (userDefinedSecretProvider is null)
-            {
-                throw new InvalidOperationException(
+            var userDefinedSecretProvider = context.HttpContext.RequestServices.GetService<ISecretProvider>() ?? throw new InvalidOperationException(
                     "Cannot retrieve the shared access key to validate the HTTP request because no Arcus secret store was registered in the application," 
                     + $"please register the secret store with '{nameof(IHostBuilderExtensions.ConfigureSecretStore)}' on the '{nameof(IHostBuilder)}' or with 'AddSecretStore' on the '{nameof(IServiceCollection)}'," 
                     + "for more information on the Arcus secret store: https://security.arcus-azure.net/features/secret-store");
-            }
-
-            Task<IEnumerable<string>> rawSecretAsync = userDefinedSecretProvider.GetRawSecretsAsync(_secretName);
-            if (rawSecretAsync is null)
-            {
-                throw new InvalidOperationException(
+            Task<IEnumerable<string>> rawSecretAsync = userDefinedSecretProvider.GetRawSecretsAsync(_secretName) ?? throw new InvalidOperationException(
                     $"Configured {nameof(ISecretProvider)} is not implemented correctly as it returns 'null' for a {nameof(Task)} value when calling {nameof(ISecretProvider.GetRawSecretAsync)}");
-            }
-
             IEnumerable<string> foundSecrets = await rawSecretAsync;
-            if (foundSecrets is null)
-            {
-                throw new SecretNotFoundException(_secretName);
-            }
-
-            return foundSecrets.ToArray();
+            return foundSecrets is null ? throw new SecretNotFoundException(_secretName) : foundSecrets.ToArray();
```

Comment:
Unrelated change that does not make it more readable.
An extra thing to consider when developing for Arcus, is that the code should be super simple and we should avoid unnecessary long lines. In this case, the most important value (the actual found secret) is hidden away at the end of the line, for example; in favor of using a syntactic sugar. Which does not make it more readable.

Reply:
This is indeed something I did not consider before creating this PR.